### PR TITLE
[RSDK-3308] Added support for reading digital interrupt pins as input

### PR DESCRIPTION
**Pull request by natalieh235**
- added struct gpioInterruptWrapperPin to wrap digital interrupts as a GPIOPin
- tested with an encoder component by reading the value of the interrupt pins while turning the encoder and ensuring they go low and high

### CODE REVIEW

**Comment by penguinland** [comment]
Try running `make lint`, and fixing the issues it notes. It looks like the main one is that `fmt.Errorf` should be replaced with `errors.New` if you're not actually using any formatting functionality. That will hopefully get the tests to run/pass.

File: components/board/genericlinux/board.go [hunk 1]
```diff
@@ -588,7 +588,14 @@ func (b *sysfsBoard) GPIOPinByName(pinName string) (board.GPIOPin, error) {
 	// Otherwise, the pins are stored in b.gpios.
 	pin, ok := b.gpios[pinName]
 	if !ok {
```
Comment (penguinland) [inline components/board/genericlinux/board.go:590]: I'd flip these if statements around. This lets you return early when you have the answer, and will keep the code less indented (which makes it easier to read/understand/maintain).

```go
if pin, ok := b.gpios[pinName]; ok {
    return pin, nil
}
// Check if the pin is a digital interrupt
if interrupt, ok := b.interrupts[pinName]; ok {
    return &gpioInterruptWrapperPin{*interrupt}, nil
}
return nil, errors.Errorf("cannot find unknown pin %s", pinName)
```

**Review by penguinland — COMMENTED** [review]
What have you done to check that this works properly?

**Review by penguinland — APPROVED** [review]
Same question as before about how you tested this. but things look very good! LGTM, and we'll do one final round of testing on hardware before we merge it in.

**Comment by natalieh235** [comment]
edited pr description to describe how this was tested

**Pull request merged**